internal/datastore/mongo_store.go

```diff
@@ -114,14 +144,86 @@ func (mongodb MongoDB) DeleteQueue(id db.QueueId) error {
 	return nil
 }
 
-func (mongodb MongoDB) AddTokenToQueue(db.QueueId, db.Token) {
-	panic("Not implemented")
+func (mongodb MongoDB) AddTokenToQueue(id db.QueueId, token db.Token) (db.TokenId, error) {
+	token.Id = ""
+	token.QueueId = id
+
+	// Lock queue to ensure 2 concurrent add tokens don't happen
+	queueMutex.Lock()
+
+	// if there were 2 concurrent calls to this function, 2 tokens might get the same number
+	max, err := mongodb.GetMaxToken(id)
+
+	if err != nil {
+		log.Fatal(err)
+		queueMutex.Unlock()
+		return token.Id, err
+	}
+
+	token.TokenNumber = max + 1
+
+	result, err := mongodb.Token.InsertOne(context.TODO(), token)
```

Comment:
We're using `context.TODO()` in a lot of places right now. My understanding is that this is a placeholder till we add the right context. What do you think about defining this globally at the top of the file or even in a separate config file so that it can be swapped out easily later on and make it easier for adding tests later on?

Reply:
Yeah makes sense. I'll do this in a follow up PR since this PR already has a lot of changes.